components/style/traversal.rs

```diff
@@ -311,11 +311,20 @@ pub fn recalc_style_at<'a, N, C>(context: &'a C,
             node.unstyle();
         }
 
+        // NOTE: This works now because we know we're running selector matching
+        // on all the children, but if that changes, this will need to be
+        // adjusted, and only cleared when we're completely sure of that!
```

Comment:
Please reference bug 1289868 here, and make a note in that bug that we'll need to deal with this when we stop unconditionally restyling descendants.